src/contexts/bridge.context.tsx

```diff
@@ -0,0 +1,118 @@
+import { BigNumber, ContractTransaction, PayableOverrides } from "ethers";
+import { createContext, FC, useContext, useEffect, useState } from "react";
+
+import { useEnvContext } from "src/contexts/env.context";
+import { useProvidersContext } from "src/contexts/providers.context";
+import { Bridge, Bridge__factory } from "src/types/contracts/bridge";
+
+interface BridgeContext {
+  bridge: (
+    token: string,
+    amount: BigNumber,
+    destinationNetwork: number,
+    destinationAddress: string,
+    overrides?: PayableOverrides & { from?: string | Promise<string> }
+  ) => Promise<ContractTransaction>;
+  claim: (
+    originalTokenAddress: string,
+    amount: BigNumber,
+    originalNetwork: string,
+    destinationNetwork: number,
+    destinationAddress: string,
+    smtProof: string[],
+    globalExitRootNum: number,
+    mainnetExitRoot: string,
+    rollupExitRoot: string
+  ) => Promise<ContractTransaction>;
+}
+
+const bridgeContextNotReadyErrorMsg = "The bridge context is not yet ready";
+
+const bridgeContext = createContext<BridgeContext>({
+  bridge: () => {
+    return Promise.reject(bridgeContextNotReadyErrorMsg);
+  },
+  claim: () => {
+    return Promise.reject(bridgeContextNotReadyErrorMsg);
+  },
+});
+
+const BridgeProvider: FC = (props) => {
+  const env = useEnvContext();
+  const { connectedProvider } = useProvidersContext();
+  const [bridgeContract, setBridgeContract] = useState<Bridge>();
+
+  const bridge = (
```

Comment:
Not for this PR, but I think that it would be useful to use object params instead of multiple arguments in the contexts as well, at the end they are like "`utils`" that we are calling from different places.

Reply:
Sure, agreed. I can change it in this PR, but I will wait till we make a decision on [this comment](https://github.com/hermeznetwork/bridge-ui/pull/27/files#r841504240
).

Reply:
Nice, going to take a look at it! But I'll do it in another PR to be able to review it easily.